### PR TITLE
Attempt to resolve ovm3 jenkins failure

### PR DESCRIPTION
**Pull request by rsafonseca**
Resolve unsafe cast warnings reported by lint checker
Add some debug config to compiler plugin

### CODE REVIEW

**Comment by bhaisaab** [comment]
Looks like you've got it @rsafonseca please squash the commits all 50+ of 'em :)

**Comment by rsafonseca** [comment]
Hi Rohit,
Sorry for spamming you guys so hard this weekend eheh :)
I've found the problem, but it's still not properly fixed. Cause seems to
be a 20k+ string in XenTest.java
Gave me some work to make that a StringBuilder instead of just adding
strings.. this should use a lot less heap and cpu, but now it seems to be
failing consistently, although build doesn't stop because i've put a flag
on the compiler plugin to continue anyway.
Will squash the whole thing when i get all SUCCESS instead of UNSTABLE ;)
Although 0 travis failures so far.. yay for the other fix hehe

**Comment by rsafonseca** [comment]
Just as i said travis was stable, got a lot of failures haha :)
Upon looking into it, travis didn't fail, it is correctly reporting that i killed ovm3 lol, so travis fix is working as expected.. this PR is the one who isn't 
Will fix soon, i can now reproduce the issue locally :)

**Comment by rsafonseca** [comment]
Victory!
One of the problem nodes just compiled it successfully and ran until the end without problems
Will now get started on cleaning up the PR and just leave the fix and a couple other minor improvements made along the way

**Comment by rsafonseca** [comment]
@bhaisaab you can go ahead and merge now :)

File: plugins/hypervisors/ovm3/pom.xml [hunk 1]
```diff
@@ -27,11 +27,31 @@
...
-</dependency>
+    <dependency>
+	<groupId>xml-apis</groupId>
+	<artifactId>xml-apis</artifactId>
+	<version>1.4.01</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.xmlrpc</groupId>
+      <artifactId>xmlrpc-client</artifactId>
+      <version>3.1.3</version>
```
Comment (karuturi) [inline plugins/hypervisors/ovm3/pom.xml]: Can we resuse the version number(cs.xmlrpc.version) from parent pom?

File: plugins/hypervisors/ovm3/pom.xml [hunk 2]
```diff
@@ -27,11 +27,31 @@
...
+    </dependency>
+    <dependency>
+	<groupId>org.apache.xmlrpc</groupId>
+	<artifactId>xmlrpc-common</artifactId>
+	<version>3.1.3</version>
```
Comment (karuturi) [inline plugins/hypervisors/ovm3/pom.xml]: Can we resuse the version number(cs.xmlrpc.version) from parent pom?

File: plugins/hypervisors/ovm3/pom.xml [hunk 3]
```diff
@@ -27,11 +27,31 @@
...
+    </dependency>
+    <dependency>
+	<groupId>org.apache.commons</groupId>
+	<artifactId>commons-lang3</artifactId>
+	<version>3.4</version>
```
Comment (karuturi) [inline plugins/hypervisors/ovm3/pom.xml]: can you add this version to parent pom and use from there?

File: plugins/hypervisors/ovm3/pom.xml [hunk 4]
```diff
@@ -27,11 +27,31 @@
...
+    </dependency>
+    <dependency>
+	<groupId>log4j</groupId>
+	<artifactId>log4j</artifactId>
+	<version>1.2.17</version>
```
Comment (karuturi) [inline plugins/hypervisors/ovm3/pom.xml]: Can we reuse cs.log4j.version from parent pom?

File: plugins/hypervisors/ovm3/pom.xml [hunk 5]
```diff
@@ -27,11 +27,31 @@
...
   </parent>
   <dependencies>
-<dependency>
-    <groupId>org.apache.xmlrpc</groupId>
-    <artifactId>xmlrpc-client</artifactId>
-    <version>3.1.3</version>
-</dependency>
+    <dependency>
+	<groupId>xml-apis</groupId>
+	<artifactId>xml-apis</artifactId>
```
Comment (karuturi) [inline plugins/hypervisors/ovm3/pom.xml]: this is a new dependency. Is this conflict resolution? If yes, can this be added to the exclusion list of the dependency?

Comment (rsafonseca) [inline plugins/hypervisors/ovm3/pom.xml]: It's not a new dependency, just an explicit declaration of something that was a dependency already.

File: plugins/hypervisors/ovm3/pom.xml (hunk 1 above)
Comment (rsafonseca) [inline plugins/hypervisors/ovm3/pom.xml]: We can and should, i just kept this one the way it was, version was already specified in the version currently in master, will update

File: plugins/hypervisors/ovm3/pom.xml (hunk 2 above)
Comment (rsafonseca) [inline plugins/hypervisors/ovm3/pom.xml]: Yes, will update

File: plugins/hypervisors/ovm3/pom.xml (hunk 3 above)
Comment (rsafonseca) [inline plugins/hypervisors/ovm3/pom.xml]: I can, will do. We should also upgrade some other components to lang3 as lang is outdated

File: plugins/hypervisors/ovm3/pom.xml (hunk 4 above)
Comment (rsafonseca) [inline plugins/hypervisors/ovm3/pom.xml]: We can, though this version is already pretty old and not likely to change, will update for consistency

**Comment by rsafonseca** [comment]
@karuturi done in a nicer way, as requested :)
FYI, the only thing that is needed to actually fix the ovm3 problem is the changes to plugins/hypervisors/ovm3/src/test/java/com/cloud/hypervisor/ovm3/objects/XenTest.java

All the rest are just minor improvements made along the testing of the problem, but none are related to actually fixing it. The minor improvements i referred to are simply getting rid of some lint checker warnings about serializable classes not having the serialversionuid and doing redundant explicit casts. Also added a couple of non declared dependencies to the pom, and that's it :)

**Comment by karuturi** [comment]
Thanks for the update @rsafonseca

**Pull request closed without merging**